integration/proxy_tunnel_strategy_test.go

```diff
@@ -0,0 +1,529 @@
+
+	conf := service.MakeDefaultConfig()
+	conf.AuthServers = append(conf.AuthServers, *authAddr)
+	conf.Token = "token"
+	conf.DataDir = t.TempDir()
+
+	conf.Auth.Enabled = false
+	conf.SSH.Enabled = false
+
+	conf.Proxy.Enabled = true
+	conf.Proxy.WebAddr.Addr = net.JoinHostPort(Loopback, strconv.Itoa(ports.PopInt()))
+	conf.Proxy.PeerAddr.Addr = net.JoinHostPort(Loopback, strconv.Itoa(ports.PopInt()))
+	conf.Proxy.PublicAddrs = append(conf.Proxy.PublicAddrs, utils.FromAddr(p.lb.Addr()))
+	conf.Proxy.DisableWebInterface = true
+
+	require.NoError(t, proxy.CreateEx(t, nil, conf))
+	p.lb.AddBackend(conf.Proxy.WebAddr)
+	require.NoError(t, proxy.Start())
+
+	p.proxies = append(p.proxies, proxy)
+}
+
+// makeNode bootstraps a new teleport node instance.
+// It connects to a proxy via a reverse tunnel going through a load balancer.
+func (p *proxyTunnelStrategy) makeNode(t *testing.T) {
+	if p.node != nil {
+		require.Fail(t, "node already initialized")
+	}
+
+	node := NewInstance(InstanceConfig{
+		ClusterName: p.cluster,
+		HostID:      uuid.New().String(),
+		NodeName:    Loopback,
+		log:         utils.NewLoggerForTests(),
+	})
+
+	conf := service.MakeDefaultConfig()
+	conf.AuthServers = append(conf.AuthServers, utils.FromAddr(p.lb.Addr()))
+	conf.Token = "token"
+	conf.DataDir = t.TempDir()
+
+	conf.Auth.Enabled = false
+	conf.Proxy.Enabled = false
+	conf.SSH.Enabled = true
+
+	require.NoError(t, node.CreateEx(t, nil, conf))
+	require.NoError(t, node.Start())
+
+	p.node = node
+}
+
+// makeDatabase bootstraps a new teleport db instance.
+// It connects to a proxy via a reverse tunnel going through a load balancer.
+func (p *proxyTunnelStrategy) makeDatabase(t *testing.T) {
+	if p.db != nil {
+		require.Fail(t, "database already initialized")
+	}
+
+	dbAddr := net.JoinHostPort(Host, strconv.Itoa(ports.PopInt()))
+
+	// setup database service
+	db := NewInstance(InstanceConfig{
+		ClusterName: p.cluster,
+		HostID:      uuid.New().String(),
+		NodeName:    Loopback,
+		log:         utils.NewLoggerForTests(),
+	})
+
+	conf := service.MakeDefaultConfig()
+	conf.AuthServers = append(conf.AuthServers, utils.FromAddr(p.lb.Addr()))
+	conf.Token = "token"
+	conf.DataDir = t.TempDir()
+
+	conf.Auth.Enabled = false
+	conf.Proxy.Enabled = false
+	conf.SSH.Enabled = false
+	conf.Databases.Enabled = true
+	conf.Databases.Databases = []service.Database{
+		{
+			Name:     p.cluster + "-postgres",
+			Protocol: defaults.ProtocolPostgres,
+			URI:      dbAddr,
+		},
+	}
+
+	_, role, err := auth.CreateUserAndRole(p.auth.Process.GetAuthServer(), p.username, nil)
+	require.NoError(t, err)
+
+	role.SetDatabaseUsers(types.Allow, []string{types.Wildcard})
+	role.SetDatabaseNames(types.Allow, []string{types.Wildcard})
+	err = p.auth.Process.GetAuthServer().UpsertRole(context.Background(), role)
+	require.NoError(t, err)
+
+	// start the process and block until specified events are received.
+	require.NoError(t, db.CreateEx(t, nil, conf))
```

Comment:
Same configuration caveat as the proxy one, `AuthServers` will contain the load balancer and the `db`'s internal (deactivated) auth server.